Core/include/Acts/Propagator/EigenStepper.hpp

```diff
@@ -73,22 +72,24 @@ class EigenStepper {
     /// @param [in] stolerance is the stepping tolerance
     ///
     /// @note the covariance matrix is copied when needed
-    template <typename parameters_t>
+    template <typename parameters_t,
+              std::enable_if_t<parameters_t::s_boundRepresentation, int> = 0>
     explicit State(std::reference_wrapper<const GeometryContext> gctx,
                    std::reference_wrapper<const MagneticFieldContext> mctx,
                    const parameters_t& par, NavigationDirection ndir = forward,
                    double ssize = std::numeric_limits<double>::max(),
                    double stolerance = s_onSurfaceTolerance)
```

Comment:
```suggestion
    template <typename charge_t>
    explicit State(std::reference_wrapper<const GeometryContext> gctx,
                   std::reference_wrapper<const MagneticFieldContext> mctx,
                   const SingleBoundTrackParameters<charge_t>& par, NavigationDirection ndir = forward,
                   double ssize = std::numeric_limits<double>::max(),
                   double stolerance = s_onSurfaceTolerance)
```
Whether a track parameter is bound or free is already encoded in its type name. I do not really see why we need to add an extra member variable here to disambiguate. By using the concrete type (with just the charge template), we can let the compiler do the selection work for us and we do not have to rely on the `std::enable_if_t` magic (which I personally always find a bit hard to follow).

Reply:
Agreed.